File: zeromq/hello_world_client.cpp
#include <zmq.hpp>
#include <string>
#include <iostream>

int main(){
    zmq::context_t context(1);
    zmq::socket_t socket(context, ZMQ_REQ);

    std::cout << "Connecting to hello world server.." << std::endl;
    socket.connect("tcp://localhost:5555");

    for(int rn=0; rn != 10; ++rn){
        zmq::message_t request(5);
        memcpy(request.data(), "Hello", 5);
        std::cout << "Sending Hello " << rn << "..." << std::endl;
        socket.send(request);
        zmq::message_t reply;
        socket.recv(&reply);
        std::string *reply_str = static_cast<std::string *>(reply.data());
        std::cout << "Received reply: " << reply_str << std::endl;
    }
    return 0;
}

File: zeromq/java/zmq-demo/src/main/java/com/saihei/demo/HelloWorldClient.java
package com.saihei.demo;
import org.zeromq.ZMQ;

public class HelloWorldClient{
    public static void main(String[] args) throws Exception{
        ZMQ.Context context = ZMQ.context(1);  //1 rep number of threads
        System.out.println("Connecting to hello world server");
        ZMQ.Socket socket = context.socket(ZMQ.REQ);
        socket.connect("tcp://localhost:5555");
        for(int rn=0; rn != 10; rn++){
            String request = "Hello";
            System.out.println("Sending Hello " + rn);
            socket.send(request.getBytes(ZMQ.CHARSET), 0);
            byte[] reply = socket.recv(0);
            System.out.println("Received " + new String(reply, ZMQ.CHARSET) + " " + rn);
        }
        socket.close();
        context.term();
    }
}
